discovery/api/v1alpha1/addon_types.go

```diff
@@ -18,18 +18,30 @@ package v1alpha1
 import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 // AddonSpec defines the desired state of Addon
 type AddonSpec struct{}
 
 // AddonStatus describes the observed state of an addon and its components.
 type AddonStatus struct {
+	// Metadata extracts information about the targeted objects based on labels.
+	// +optional
+	Metadata []Metadata `json:"metadata,omitempty"`
```

Comment:
Is this `Metadata`-per-component? If so, I think it makes more sense for it to be a field of `RichReference`.